Provide an importable module

rschamp opened the pull request:
Allows this from other packages:

``` javascript
const ScratchBlocksVertical = require('scratch-blocks');
const ScratchBlocksHorizontal = require('scratch-blocks/dist/horizontal');
```

How it works:
When another package requires `scratch-blocks`, webpack looks at the package.json and finds the `webpack` entry pointing to `shim/index.js`. 
- `shim/index.js` exports horizontal and vertical Blockly from `msg/messages.js`
  - This requires `goog` from `shim/blockly_compressed_[horizontal,vertical].goog`
    - `shim/blockly_compressed_[horizontal,vertical].goog` exports the `goog` var from the actual `blockly_compressed_[horizontal,vertical]`
  - `Blockly` is provided from `shim/blocks_compressed_[horizontal,vertical].Blockly`
- `shim/blocks_compressed_[horizontal,vertical]` exports `Blockly` from `blocks_compressed`, but the `Blockly` provided to `blocks_compressed` is the one exported by `shim/blockly_compressed_[horizontal,vertical].Blockly`
- `shim/blockly_compressed_[horizontal,vertical].Blockly` exports `Blockly` from `shim/blockly_compressed_[horizontal,vertical]`
- `shim/blockly_compressed_[horizontal,vertical]` exports `{goog: goog, Blockly: Blockly}` from `blockly_compressed_[horizontal,vertical]`

The intermidate shims are required due to the way the imports and exports loaders work — you can't do a multi-level loading, or this would be simpler to do.  You can also set this up more simply in webpack (see first commit), but then you would need to build something for other packages to import. This makes a persistent shim that just exports the product of the compressed files (and messages.js).

Other packages don't need to know about the details of this shimming process: imports-loader and exports-loader will be provided by this package when webpack requires the modules out of this package.

Commit:
Provide a webpack-importable module on install

webpack.config.js sets up a cascade of imports so that the Blockly object
in blocks_compressed_horizontal and blocks_compressed_vertical is retrieved
from blocks_compressed, which in turn retrieves its Blockly object from
blockly_compressed_horizontal or blockly_compressed_vertical, respectively
(via blocks_compressed_shim_vertical and blocks_compressed_horizontal).

This is finally compiled into a single module at export.js, which is seen as
the main module by webpack when this package is imported.

Usage:
```javascript
const ScratchBlocksVertical = require('scratch-blocks').Vertical;
```

Commit:
Provide a shim that doesn't need to be built

Because this package installs imports-loader and exports-loader, when another package requires scratch-blocks, webpack will use this package's loaders to do the shimming. This way it will be built by the parent package.

tmickel reviewed (approved):
Wow cool! I'm excited, looks great to me!

Commit:
Build separate vertical and horizontal modules

This allows separate imports for vertical and horizontal libraries, and gives a way to import the library outside of webpack.

rschamp commented:
I decided it would be better if we provided a module that could be imported without webpack (e.g., for tests).  If you want to you can still import the unbuilt module, but for efficiency, the vertical library is the default imported when you `require('scratch-blocks')`.

thisandagain commented:
Looks great. Thanks @rschamp
